src/Blocks/BlockPiston.cpp

```diff
@@ -176,6 +176,13 @@ bool cBlockPistonHandler::CanPushBlock(
 				// When it's not possible for a direction, then fail
 				return false;
 			}
+
+			// Terracotta and leaves can be pushed, but not pulled - so can't stick
+			const auto StickingBlockType = a_World.GetBlock(a_BlockPos + testDir);
+			if (IsBlockMaterialTerracotta(StickingBlockType) || IsBlockMaterialLeaves(StickingBlockType))
+			{
+				return false;
```

Comment:
Doesn't this mean slime blocks can no longer push terracotta? So piston -> slime -> terracotta doesn't extend?

Reply:
It is pushed ok. It's not pulled back though. Working on it

Reply:
Done.
Do I revert the const qualifier on all of the functions or just cBlockInfo members?

Reply:
Done.